### PR TITLE
Added autoincrement table. Added readable id to recipient.

### LINKED ISSUE #242: Backend - Recipient and Connections should receive a short-readable ID when added
Add a new property to recipients where a readable id is given upon registration. Connections should receive this id from recipient.

### PR DESCRIPTION
**Pull request by joffe97**
Fix #242 
Decided to use numbers instead of a combination of both numbers and letters. This combination would add some complexity with not too much of a difference.

Case with about 200 families:
- Only numbers: 
  - Combination of 3 letters
- Both letters and numbers
  - Combination of 2 letters

Case with 1024+ families:
- Only numbers
  - Combination of 4 letters (e.g. 2595)
- Both letters and numbers
  - Combination of 3 letters (e.g. B5Q)

### CODE REVIEW

File: GiEnJul/Features/AutoIncrementRepository.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                context.UserHeaders["If-Match"] = result.ETag;
+
+                try
+                {
+                    await InsertOrReplaceAsync(result, context);
+                    return result.Value.ToString();
+                }
+                catch (StorageException e)
+                {
+                    attempts += 1;
```
Comment (RonHaland) [inline GiEnJul/Features/AutoIncrementRepository.cs:63]: Could use 'await Task.Delay(50);' here

**Review by Sommerli — COMMENTED** [review]
Comments on naming and retry-behaviour

File: GiEnJul/Features/AutoIncrementRepository.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    if (e.RequestInformation.HttpStatusCode != 412 || attempts > 3)
+                    {
+                        throw(e);
+                    }
```
[the remark this answers is not in the document]
Comment (RonHaland) [reply GiEnJul/Features/AutoIncrementRepository.cs:67]: What we do with each retry is:
1. Get value
2. increment
3. insert value
I don't think the retryPolicy will handle the entire operation?

A similar solution would be to wrap it in some sort of scope/context with a retry policy, but I'm not very familiar with how to use that.

[the remark this answers is not in the document]
Comment (Sommerli) [reply GiEnJul/Features/AutoIncrementRepository.cs:67]: > I don't think the retryPolicy will handle the entire operation?
makes sense, then the solution should work fine as is

File: GiEnJul/Models/Recipient.cs [hunk 3]
```diff
@@ -12,7 +12,9 @@ public class Recipient
         public string Note { get; set; }
         public string EventName { get; set; }
         public string Location { get; set; }
-        
+        public string InternalId { get; set; }
```
[the remark this answers is not in the document]
Comment (joffe97) [reply GiEnJul/Models/Recipient.cs:15]: Agree! I'll change to familyId.

File: GiEnJul/Features/AutoIncrementRepository.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        public AutoIncrementRepository(
+            ISettings settings, 
+            IMapper mapper, 
+            ILogger log, 
+            string tableName = "AutoIncrement"
+        ) : base(settings, tableName, mapper, log)
+        { }
+
+        public DateTimeOffset GetTimestamp { get; private set; }
```
[the remark this answers is not in the document]
Comment (joffe97) [reply GiEnJul/Features/AutoIncrementRepository.cs:27]: Nope it's not 👍

**Review by RonHaland — APPROVED** [review]

**Pull request merged**